kani-compiler/src/kani_middle/attributes.rs

```diff
@@ -540,6 +540,13 @@ impl<'tcx> KaniAttributes<'tcx> {
         assert!(matches!(
                 &harness.kind, HarnessKind::ProofForContract { target_fn }
                 if *target_fn == name.to_string()));
+
+        // Only emit an error if we are trying to actually verify the contract.
+        // (If we are running the list subcommand, we just report later that there are no contracts for this harness).
+        if is_list_enabled {
+            return;
+        }
```

Comment:
People may want to write harnesses before they write the contracts. For instance, if I'm planning to write a contract for a function, I could write:

```rust
fn foo(x: usize) -> usize { ... }

#[kani::proof_for_contract(foo)
fn harness() {
  let x: usize = kani::any();
  foo(x)
}
```

It makes sense to error during verification, since we can't verify a contract that doesn't exist. But I didn't think it made sense to punish people for being halfway done; they may want to use the list command to see what work they have left to do.

That being said, I'm not even sure I could make this distinction if we're making the compiler unaware of the list command, so we may have to make it a compiler error.